pkg/oauth/server/osinserver/registrystorage/storage.go

```diff
@@ -248,6 +255,20 @@ func (s *storage) convertToAccessToken(data *osin.AccessData) (*oauthapi.OAuthAc
 	if err := s.user.ConvertToAccessToken(data.UserData, token); err != nil {
 		return nil, err
 	}
+
+	// Check if the optional inactivity Timeout needs to be set
+	var timeoutsin int32 = 0
+	wrapper := data.Client.(*clientWrapper)
+	inactivityTimeout := wrapper.GetTokenTimeoutSeconds()
+	if inactivityTimeout != nil {
+		timeoutsin = *inactivityTimeout
+	} else if s.tokentimeout != nil {
+		timeoutsin = *s.tokentimeout
+	}
+	if timeoutsin > 0 {
```

Comment:
Seems like validation logic that should exist in master config + OAuth client code.

Reply:
actually, timeoutsin = 0 is just fine here, I will just remove the conditional

Reply:
done